iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/vector.inl

```diff
@@ -358,7 +349,7 @@ inline bool vector<T, Capacity>::erase(iterator position) noexcept
 {
     if ((begin() <= position) && (position < end()))
     {
-        uint64_t index{static_cast<uint64_t>(position - begin()) % (sizeof(element_t) * Capacity)};
+        uint64_t index{static_cast<uint64_t>(position - begin()) % sizeof(decltype(m_data))};
```

Comment:
I am not entirely sure, but I think this is dangerous if `UninitializedArray<...>`  (which is the declared type) contains additional data (and it might in the future, causing a subtle bug here).

Need to confirm or refute this and if confirmed solve it differently.

Reply:
Since the `UninitializedArray` should remain a very low-level construct, there should be no additional data in future but I added the test `SizeOfUninitializedArrayEqualsCStyleArray` which should reveal such changes. Do you think this is sufficient?

Reply:
After a further review and considering the iterator/pointer type I think the modulo operation is and was not necessary at all.

Reply:
@FerdinandSpitzschnueffler If removing the % operation as suggested works and does not brake anything unforeseen, no further test is necessary and the issue can be resolved.